src/analysis.jl

```diff
@@ -25,47 +25,45 @@ function code(c::DynamicCall; optimize = false)
   codeinfo = code_typed(c.f, argtypes(c), optimize = optimize)
   @assert length(codeinfo) == 1
   codeinfo = codeinfo[1]
-  linearize!(codeinfo[1])
+  # linearize!(codeinfo[1])
   return codeinfo
 end
 
-struct StaticCall <: Call
-  method_instance::MethodInstance
-end
-
-argtypes(c::StaticCall) = Tuple{c.method_instance.specTypes.parameters[2:end]...}
-types(c::StaticCall) = c.method_instance.specTypes
-method(c::StaticCall) = c.method_instance.def
-
-method_expr(c::StaticCall) = method_expr(method(c).name, argtypes(c))
-
-function code(c::StaticCall; optimize = false)
-  # TODO static call graph can only be computed with optimize=true, so analyzing with optimized=false will skip inlined methods
-  codeinfo = get_code_info(c.method_instance, optimize=true)
-  linearize!(codeinfo[1])
-  return codeinfo
-end
+# struct StaticCall <: Call
+#   method_instance::MethodInstance
+# end
+#
+# argtypes(c::StaticCall) = Tuple{c.method_instance.specTypes.parameters[2:end]...}
+# types(c::StaticCall) = c.method_instance.specTypes
+# method(c::StaticCall) = c.method_instance.def
+#
+# method_expr(c::StaticCall) = method_expr(method(c).name, argtypes(c))
+#
+# function code(c::StaticCall; optimize = false)
+#   # TODO static call graph can only be computed with optimize=true, so analyzing with optimized=false will skip inlined methods
+#   codeinfo = get_code_info(c.method_instance, optimize=true)
+#   linearize!(codeinfo[1])
+#   return codeinfo
+# end
 
 function eachline(f, code, line = -1)
-  for l in code.code
-    if l isa LineNumberNode
-      line = l.line
-    else
-      f(line, l)
-    end
+  for (i, l) in enumerate(code.code)
+    line = code.linetable[code.codelocs[i]].line
+    f(line, l)
   end
 end
 
 exprtype(code, x) = typeof(x)
-exprtype(code, x::Expr) = x.typ
+exprtype(code, x::Expr) = Union{}
+exprtype(code, x::Core.TypedSlot) = x.typ
 exprtype(code, x::QuoteNode) = typeof(x.value)
-exprtype(code, x::SSAValue) = code.ssavaluetypes[x.id+1]
-exprtype(code, x::SlotNumber) = code.slottypes[x.id]
+exprtype(code, x::Core.SSAValue) = code.ssavaluetypes[x.id+1]
+# exprtype(code, x::Core.SlotNumber) = code.slottypes[x.id]
 
 rebuild(code, x) = x
 rebuild(code, x::QuoteNode) = x.value
-rebuild(code, x::Expr) = Expr(x.head, rebuild.(code, x.args)...)
-rebuild(code, x::SlotNumber) = code.slotnames[x.id]
+rebuild(code, x::Expr) = Expr(x.head, rebuild.(Ref(code), x.args))
```

Comment:
Wouldn't this still need to be splatted?

Reply:
Yeah, seems that way. I thought there was a `Expr(head, args)` constructor too.